### PR TITLE
add spring cloud auth to AuthManager

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java [hunk 1]
```diff
@@ -76,6 +77,17 @@ public Azure getAzureClient(String sid) throws IOException {
         return azure;
     }
 
+    public AppPlatformManager getAzureSpringCloudClient(String sid) throws IOException {
+        if (getAzureManager() == null) {
+            throw new IOException(CANNOT_GET_AZURE_MANAGER);
+        }
+        AppPlatformManager client = getAzureManager().getAzureSpringCloudClient(sid);
+        if (client == null) {
+            throw new IOException(String.format(CANNOT_GET_AZURE_BY_SID, sid));
```
Comment (Flanker32) [inline Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java]: throw the exception in `getAzureSpringCloudClient` method?

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+        final List<ServiceResourceInner> clusters = new ArrayList<>();
+        List<Subscription> subs = AzureMvpModel.getInstance().getSelectedSubscriptions();
+        if (subs.size() == 0) {
+            return clusters;
+        }
+        Observable.from(subs).flatMap((sd) -> Observable.create((subscriber) -> {
+            try {
+                List<ServiceResourceInner> clustersInSubs = listAllSpringCloudClustersBySubscription(
+                        sd.subscriptionId());
+                synchronized (clusters) {
```
Comment (Flanker32) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java]: use `CopyOnWriteArrayList` ?

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+        return getSpringManager(getSubscriptionId(appId)).apps().inner().get(getResourceGroup(appId),
+                getClusterName(appId), getAppName(appId));
+    }
+
+    public static DeploymentResourceInner getActiveDeploymentForApp(String appId) throws IOException {
+        String sid = getSubscriptionId(appId);
+        String rid = getResourceGroup(appId);
+        String cid = getClusterName(appId);
+        String appName = getAppName(appId);
+        AppResourceInner app = getSpringManager(sid).apps().inner().get(rid, cid, appName);
```
Comment (Flanker32) [inline Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java]: use `getAppByName` and `getAppDeployment` here

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java (hunk 2 above)
Comment (andxu) [reply Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java]: CopyOnWriteArrayList is used in the scenarios where the read operations.

File: Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java (hunk 3 above)
Comment (andxu) [reply Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/springcloud/AzureSpringCloudMvpModel.java]: Done

File: Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java (hunk 1 above)
Comment (andxu) [reply Utils/azuretools-core/src/com/microsoft/azuretools/authmanage/AuthMethodManager.java]: Done

**Pull request merged**